### PR TITLE
KATI-13_Patient-Profile-View-Feature

### PR DESCRIPTION
**Pull request by vtnlinh95**

### CODE REVIEW

File: app/src/main/java/com/kms/cura/utils/PatinetProfileUtils.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
+package com.kms.cura.utils;
+
+import com.kms.cura.entity.user.PatientUserEntity;
+
+import java.sql.Date;
+
+/**
+ * Created by linhtnvo on 6/8/2016.
+ */
+public class PatinetProfileUtils {
```
Comment (hieumai) [inline app/src/main/java/com/kms/cura/utils/PatinetProfileUtils.java]: Typo. Rename this class

Comment (hieumai) [inline app/src/main/java/com/kms/cura/utils/PatinetProfileUtils.java]: Also, if this class is to contain the current user then change it to something appropriately

File: com.kms.cura.entity/src/com/kms/cura/entity/user/PatientUserEntity.java [hunk 2]
```diff
@@ -21,7 +24,10 @@ public PatientUserEntity(String id, String name, String email, String password,
 	}
 
 	public String getGender() {
-		return gender;
+		if("M".equals(gender)){
+			return "Male";
+		}
+		return "Female";
```
Comment (hieumai) [inline com.kms.cura.entity/src/com/kms/cura/entity/user/PatientUserEntity.java]: Please extract "M" into constant. "Male" and "Female" should be put into resources file

File: app/src/main/java/com/kms/cura/utils/PatinetProfileUtils.java (hunk 1 above)
Comment (hieumai) [inline app/src/main/java/com/kms/cura/utils/PatinetProfileUtils.java]: Also, consider the case to reuse for Doctor also

File: app/src/main/java/com/kms/cura/utils/PatientProfileCurrent.java [hunk 3]
```diff
@@ -0,0 +1,31 @@
+package com.kms.cura.utils;
+
+import com.kms.cura.entity.user.PatientUserEntity;
+
+/**
+ * Created by linhtnvo on 6/8/2016.
+ */
+public class PatientProfileCurrent {
```
Comment (hieumai) [inline app/src/main/java/com/kms/cura/utils/PatientProfileCurrent.java]: So this class only save current patient ? How about doctor or admin ? And how about when you need to check the current user type ?

File: com.kms.cura.entity/src/com/kms/cura/entity/user/PatientUserEntity.java [hunk 4]
```diff
@@ -21,7 +25,10 @@ public PatientUserEntity(String id, String name, String email, String password,
 	}
 
 	public String getGender() {
-		return gender;
+		if(gender.equals(GENDER_MALE)){
+			return "Male";
+		}
+		return "Female";
```
Comment (hieumai) [inline com.kms.cura.entity/src/com/kms/cura/entity/user/PatientUserEntity.java]: Again, your "Male" and "Female" strings should go into resources file

File: app/src/main/java/com/kms/cura/utils/PatientProfileCurrent.java (hunk 3 above)
Comment (vtnlinh95) [inline app/src/main/java/com/kms/cura/utils/PatientProfileCurrent.java]: @hieumai  I will implement doctor in the branch of doctor. Why I need to check the current user type ???

File: com.kms.cura.entity/src/com/kms/cura/entity/user/PatientUserEntity.java (hunk 4 above)
Comment (vtnlinh95) [inline com.kms.cura.entity/src/com/kms/cura/entity/user/PatientUserEntity.java]: @hieumai  I don't get it. The R.string. can not call outside the android project

Comment (hieumai) [inline com.kms.cura.entity/src/com/kms/cura/entity/user/PatientUserEntity.java]: Then you should create enums for Male and Female. Pass that to the android app side to get the correct label.

File: app/src/main/java/com/kms/cura/utils/PatientProfileCurrent.java (hunk 3 above)
Comment (hieumai) [inline app/src/main/java/com/kms/cura/utils/PatientProfileCurrent.java]: This app will never need to check for current user type ? Are you sure ?

**Comment by vtnlinh95** [comment]
@hieumai  Fixed

**Comment by hieumai** [comment]
OK from me. @huuduy216  and @toannguyen71195  please start verifying.

**Comment by vtnlinh95** [comment]
@toannguyen71195 @huuduy216  wait for me to rebase the branch

**Comment by toannguyen71195** [comment]
@hieumai Verified

**Comment by huuduy216** [comment]
verified okay

**Comment by hieumai** [comment]
@vtnlinh95 please solve conflict and squash this pull request

**Comment by hieumai** [comment]
@vtnlinh95 also please re-use the CurrentUser on this

File: app/src/main/java/com/kms/cura/utils/CurrentUserProfile.java [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+    public void setData(UserEntity src) {
+        this.entity = src;
+    }
+
+    public UserEntity getEntity() {
+        return entity;
+    }
+
+    public boolean isDoctor() {
+        if (entity instanceof DoctorUserEntity) {
```
Comment (hieumai) [inline app/src/main/java/com/kms/cura/utils/CurrentUserProfile.java]: Just `return entity instanceof DoctorUserEntity` here would be enough

File: app/src/main/java/com/kms/cura/utils/CurrentUserProfile.java [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+            return true;
+        }
+        return false;
+    }
+
+    public boolean isPatient() {
+        if (entity instanceof PatientUserEntity) {
```
Comment (hieumai) [inline app/src/main/java/com/kms/cura/utils/CurrentUserProfile.java]: Just `return entity instanceof PatientUserEntity` here would be enough

File: app/src/main/java/com/kms/cura/view/activity/PatientProfileViewActivity.java [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+        } else {
+            txtDOB = loadText(entity.getBirth().toString(), R.id.txtDOB);
+        }
+        txtLocation = loadText(entity.getLocation(), R.id.txtLocation);
+        txtInsurance = loadText(entity.getInsurance(), R.id.txtInsurance);
+        txtHealthConcerns = loadText(entity.getHealthConcern(), R.id.txtHealthConcern);
+    }
+
+    public String getGender(PatientUserEntity patientUserEntity){
+        if(patientUserEntity.getGender() == null){
```
Comment (hieumai) [inline app/src/main/java/com/kms/cura/view/activity/PatientProfileViewActivity.java]: re-format code here

File: app/src/main/java/com/kms/cura/view/activity/PatientProfileViewActivity.java [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+        if (src == null) {
+            textView.setHeight(0);
+        } else {
+            String newString = null;
+            try {
+                newString = new String(src.getBytes("ISO-8859-1"),"UTF-8");
+            } catch (UnsupportedEncodingException e) {
+                textView.setText(src);
+            }
+            textView.setText(newString);
```
Comment (hieumai) [inline app/src/main/java/com/kms/cura/view/activity/PatientProfileViewActivity.java]: So if UnsupportedEncodingException happens, still set newString (which is null) to textView ?

Comment (vtnlinh95) [inline app/src/main/java/com/kms/cura/view/activity/PatientProfileViewActivity.java]: @hieumai  I will set textview with the text getting from the server.

**Comment by vtnlinh95** [comment]
@hieumai fixed

**Comment by hieumai** [comment]
OK. @vtnlinh95 squash the commit so I can merge

**Pull request merged**